std/json.d

```diff
@@ -346,10 +346,11 @@ struct JSONValue
             type_tag = JSON_TYPE.STRING;
             store.str = arg;
         }
-        else static if(is(T : char[]))
+        else static if(isNarrowString!T)
         {
+            import std.utf : byUTF;
             type_tag = JSON_TYPE.STRING;
-            store.str = arg.idup;
+            store.str = byUTF!char(arg).array.dup;
```

Comment:
I know. But wouldn't work without it.

Even though the result from "array" is unique, I can't seem to make it "immutable" in safe code. Please help?